doc/manual/cookbooks/inner_core_convection/inner_core_convection_assembler.cc

```diff
@@ -0,0 +1,86 @@
+template <int dim>
+class PhaseBoundaryAssembler :
```

Comment:
I would prefer creating the lib out of 2 .cc files and just reference that file instead of copying it here.